packages/components/src/utils/getTextAsHtml.ts

```diff
@@ -20,31 +22,79 @@ export const getTextAsHtml = (content?: (HardBreakProps | TextProps)[]) => {
     return ""
   }
 
-  return content
-    .map((node) => {
-      if (node.type === "hardBreak") {
-        return "<br />"
-      }
+  const output: string[] = []
+  let existingLinkMark: Marks | undefined = undefined
+
+  // At every step, we will close off all marks except for links
+  // First encounter with a link, always open it first before other marks
+  // Close all other marks first before closing the link mark
+  content.forEach((node) => {
+    if (node.type === "hardBreak") {
+      output.push("<br />")
+      return
+    }
+
+    const newLinkMark = node.marks?.find((mark) => mark.type === "link")
+    const isLinkMarkNew =
+      !!newLinkMark && !isEqual(existingLinkMark, newLinkMark)
+
+    // Close off the existing link mark if it is different
+    if (isLinkMarkNew && !!existingLinkMark) {
+      output.push(`</${MARK_DOM_MAPPING.link}>`)
+      existingLinkMark = undefined
+    }
 
-      if (!node.marks) {
-        return node.text
-      }
+    // If there are no marks, just push the text
+    if (!node.marks) {
+      output.push(node.text)
+      return
+    }
+
+    if (isLinkMarkNew) {
+      existingLinkMark = newLinkMark
+
+      // Sort such that the link mark is the first item
+      node.marks.sort((a, b) => {
+        if (a.type === "link") {
+          return -1
+        }
+
+        return 1
+      })
 
-      let output = node.text
       node.marks.forEach((mark) => {
         if (mark.type === "link") {
-          output = `<${MARK_DOM_MAPPING[mark.type]} href="${
-            mark.attrs.href
-          }">${output}</${MARK_DOM_MAPPING[mark.type]}>`
-          return
+          output.push(
+            `<${MARK_DOM_MAPPING[mark.type]} target="${mark.attrs.target ?? "_self"}" href="${mark.attrs.href}">`,
```

Comment:
nit: we can just push the type directly (`MARK_DOM_MAPPING.link` rather than go through 1. more leel of indirection

Reply:
no real actionable but the `??` operator doesn't handle the empty string case (in case we have such a case)

Reply:
Updated in 4127bada6311bf562b2500609445d16bbce877dc.